crates/pathfinder/src/storage/state.rs

```diff
@@ -201,6 +221,22 @@ impl GlobalStateTable {
 
         row.transpose()
     }
+
+    /// Retrieves the latest global StarkNet state from the [GlobalStateTable]. Latest is defined as the
+    /// record with the largest [StarknetBlockNumber].
+    pub fn get_latest_state(
+        transaction: &Transaction,
+    ) -> anyhow::Result<Option<GlobalStateRecord>> {
+        Self::state_query_row(
+            transaction,
+            r"SELECT * FROM global_state
+                    NATURAL JOIN ethereum_transactions
+                    NATURAL JOIN ethereum_blocks
+                    ORDER BY starknet_block_number DESC
```

Comment:
Wrong line for comment but could add a TODO around these/migration parts to index the starknet_block_number, and the columns used for the join. 

On `NATURAL JOIN` I've never used this syntax in favor of being explicit on the column I'm joining (`JOIN ethereum_transactions eth_tx ON (global_state.X = eth_tx.Y)`) as these columns then become greppable, and more importantly don't require remembering anything about the structure.

Reply:
Revolved offline; will take another look into naming columns explicitly.